File: SRP/Example/MarsRover/MarsRoverKata/Commands/TurnRightCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MarsRoverKata.Commands
{
    public class TurnRightCommand : ICommand
    {
        public Rover Rover { get; set; }

        public TurnRightCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.TurnRight();
        }
    }
}

File: LSP+ISP/Java/MarsRoverKata/src/main/java/MarsRoverKata/Commands/TurnRightCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Rover;

public class TurnRightCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public TurnRightCommand(Rover rover) {
        this.rover = rover;
    }

    public boolean execute() {
        return rover.turnRight();
    }

}
